pkg/commonvpc/lint_test_functionality.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commonvpc
+
+import (
+	"fmt"
+	"testing"
+
+	"path/filepath"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/linter"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+type VpcLintTest struct {
+	VpcTestCommon
+	Enable        []string
+	Disable       []string
+	PrintAllLints bool
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// lint:
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+const lintOut = "lint_out"
+
+func (tt *VpcLintTest) TestSingleLint(t *testing.T, rc ResourcesContainer) {
+	// all tests in lint mode
+	// output use case is not significant here, but being used so that lint test can rely on existing mechanism
+	tt.UseCases = []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints}
+	// init test - set the input/output file names according to test name
+	tt.initTest()
+
+	// get vpcConfigs obj from parsing + analyzing input config file
+	vpcConfigs := tt.getVPCConfigs(t, tt.InputConfig, rc)
+
+	// generate actual output for all use cases specified for this test
+	err := tt.runLintTestPerUseCase(t, vpcConfigs.Configs(), lintOut)
+	require.Equal(t, tt.ErrPerUseCase[vpcmodel.AllEndpoints], err, "comparing lint actual err to expected err")
+	for uc, outFile := range tt.ActualOutput {
+		fmt.Printf("lint test %s use-case %d - generated output file: %s\n", tt.Name, uc, outFile)
+	}
+}
+
+// runExplainTestPerUseCase executes lint for the required use case and compares/generates the output
+func (tt *VpcLintTest) runLintTestPerUseCase(t *testing.T, cConfigs map[string]*vpcmodel.VPCConfig, outDir string) error {
```

Comment:
Function name is confusing - lint doesn't really have use cases.
By the way, why isn't "lint" a separate use case?

Reply:
1. Agreed, I'll rename
2. In the `linter` we used another flow.  One reason for not using the `usecases` flow is that in lint we sometimes analyze all the configs together and not one by one as in the `usecases` flow. Do you want me to open an issue for discussing (and possibly implementing) this?

Reply:
yes, let's have an issue to discuss lint as a separate usecase

Reply:
https://github.com/np-guard/vpc-network-config-analyzer/issues/887